source/v5/mqtt5_encoder.c

```diff
@@ -800,7 +800,25 @@ static int s_aws_mqtt5_encoder_begin_publish(struct aws_mqtt5_encoder *encoder,
 
         local_publish_view.topic = outbound_topic;
         if (outbound_topic_alias != 0) {
+            AWS_LOGF_DEBUG(
+                AWS_LS_MQTT5_GENERAL,
+                "(%p) mqtt5 client encoder - PUBLISH packet using topic alias value %" PRIu16,
+                (void *)encoder->config.client,
+                outbound_topic_alias);
+            if (outbound_topic.len == 0) {
+                AWS_LOGF_DEBUG(
+                    AWS_LS_MQTT5_GENERAL,
+                    "(%p) mqtt5 client encoder - PUBLISH packet dropping topic field for previously established alias",
+                    (void *)encoder->config.client);
+            }
             local_publish_view.topic_alias = &outbound_topic_alias;
+        } else {
```

Comment:
Trivial: We could probably do without this log. I may be missing something but I don't think it's providing any useful information and the absence of the topic alias log above should be enough to infer that the publish isn't using topic aliasing. Also unsure whether we need to NULL

I think local_publish_view.topic_alias = NULL may be unnecessary. if outbound_topic_alias == 0, it's already NULL or an error was raised by the resolver. Could be missing something deeper tho.

Reply:
null assignment is necessary; up until now we're using what is specified by the user which could be an invalid mapping.

Reply:
As an addendum, we're always in a better position if the logging tells us exactly what happened on its own, rather than having to read the log and the code and infer what happened based on what didn't show up in the logs.